src/Grpc/JsonTranscoding/src/Shared/ServiceDescriptorHelpers.cs

```diff
@@ -444,6 +477,23 @@ static void RecursiveVisitMessages(Dictionary<string, FieldDescriptor> queryPara
         }
     }
 
+    private static bool IsCustomType(MessageDescriptor messageDescriptor)
+    {
+        // The messages flags here should be kept in sync with GrpcDataContractResolver.TryCustomizeMessage.
+        if (IsWrapperType(messageDescriptor) ||
+            messageDescriptor.FullName == Timestamp.Descriptor.FullName ||
+            messageDescriptor.FullName == Duration.Descriptor.FullName ||
+            messageDescriptor.FullName == FieldMask.Descriptor.FullName ||
+            messageDescriptor.FullName == Struct.Descriptor.FullName ||
+            messageDescriptor.FullName == ListValue.Descriptor.FullName ||
+            messageDescriptor.FullName == Value.Descriptor.FullName ||
+            messageDescriptor.FullName == Any.Descriptor.FullName)
```

Comment:
do we also need to handle "wrappers", i.e. `FloatValue`, `Int32Value`, etc? `NullValue`? `Empty`? [citation](https://protobuf.dev/programming-guides/proto3/#json)

Reply:
Wrappers are handled in this PR.